test/integ/test_put_get_with_gcp_account.py

```diff
@@ -322,3 +361,133 @@ def test_auto_compress_off_gcp(tmpdir, conn_cnx, db_parameters):
+
+
+@pytest.mark.parametrize('error_code', [401, 403, 408, 429, 500, 503])
+def test_get_gcp_file_object_http_recoverable_error_refresh_with_downscoped(tmpdir, conn_cnx, db_parameters, error_code):
+    fname = str(tmpdir.join('test_put_get_with_gcp_token.txt.gz'))
+    original_contents = "123,test1\n456,test2\n"
+    with gzip.open(fname, 'wb') as f:
+        f.write(original_contents.encode(UTF8))
+    tmp_dir = str(tmpdir.mkdir('test_put_get_with_gcp_token'))
+    table_name = random_string(5, 'snow32807_')
+
+    with conn_cnx() as cnx:
+        with cnx.cursor() as csr:
+            try:
+                csr.execute('ALTER SESSION SET GCS_USE_DOWNSCOPED_CREDENTIAL = TRUE')
+            except ProgrammingError:
+                # skip test if parameter is not available (server not updated)
+                return
+            csr.execute("create or replace table {} (a int, b string)".format(table_name))
+            try:
+                from requests import put, get, head
+
+                def mocked_put(*args, **kwargs):
+                    if mocked_put.counter == 0:
+                        exc = requests.exceptions.HTTPError(response=requests.Response())
+                        exc.response.status_code = error_code
+                        mocked_put.counter += 1
+                        raise exc
+                    else:
+                        return put(*args, **kwargs)
+
+                mocked_put.counter = 0
+
+                def mocked_head(*args, **kwargs):
+                    if mocked_head.counter == 0:
+                        mocked_head.counter += 1
+                        exc = requests.exceptions.HTTPError(response=requests.Response())
+                        exc.response.status_code = error_code
+                        raise exc
+                    else:
+                        return head(*args, **kwargs)
+
+                mocked_head.counter = 0
+
+                def mocked_file_agent(*args, **kwargs):
+                    agent = SnowflakeFileTransferAgent(*args, **kwargs)
+                    agent.renew_expired_client = mock.MagicMock(
+                        wraps=agent.renew_expired_client
+                    )
+                    mocked_file_agent.agent = agent
+                    return agent
+
+                with mock.patch('snowflake.connector.cursor.SnowflakeFileTransferAgent',
+                                side_effect=mocked_file_agent):
+                    with mock.patch('requests.put', side_effect=mocked_put):
+                        with mock.patch('requests.head', side_effect=mocked_head):
+                            csr.execute("put file://{} @%{} auto_compress=true parallel=30".format(fname, table_name))
+                    if error_code == 401:
+                        assert mocked_file_agent.agent.renew_expired_client.call_count == 2
+                assert csr.fetchone()[6] == 'UPLOADED'
+                csr.execute("copy into {}".format(table_name))
```

Comment:
You could use `PURGE=True` here instead of RMing the file manually in the next execute